File: python/tripservice_mechanical/TripService.py
from TripDAO import TripDAO
from UserSession import UserSession
from UserNotLoggedInException import UserNotLoggedInException

class TripService:
  def getTripsByUser(self, user):
    logged_user = UserSession.getInstance().getLoggedUser()
    isFriend = False
    tripList = []
    if logged_user:
      for friend in user.getFriends():
        if friend is logged_user:
          isFriend = True
          break;
      if isFriend:
        tripList = TripDAO.findTripsByUser(user)
      return tripList
    else:
      raise UserNotLoggedInException()

File: php/src/TripServiceKata/Trip/TripService.php
<?php

namespace TripServiceKata\Trip;

use TripServiceKata\User\User;
use TripServiceKata\User\UserSession;
use TripServiceKata\Exception\UserNotLoggedInException;

class TripService
{
    /**
     * @return Trip[]
     * @throws UserNotLoggedInException
     */
    public function getTripsByUser(User $user): array
    {
        $tripList = array();
        $loggedUser = UserSession::getInstance()->getLoggedUser();
        $isFriend = false;
        if ($loggedUser != null) {
            foreach ($user->getFriends() as $friend) {
                if ($friend == $loggedUser) {
                    $isFriend = true;
                    break;
                }
            }
            if ($isFriend) {
                $tripList = TripDAO::findTripsByUser($user);
            }
            return $tripList;
        } else {
            throw new UserNotLoggedInException();
        }
    }
}
